addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/XMLUtils.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol.xml;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.Zone;
+import org.openhab.binding.yamahareceiver.internal.protocol.InputWithPlayControl;
+import org.openhab.binding.yamahareceiver.internal.protocol.ReceivedMessageParseException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * Utility methods for XML handling
+ *
+ * @author David Graeff - Initial contribution
+ *
+ */
+public class XMLUtils {
+    // We need a lot of xml parsing. Create a document builder beforehand.
+    static final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+    static Node getNode(Node parent, String[] nodePath, int offset) {
+        if (parent == null) {
+            return null;
+        }
+        if (offset < nodePath.length - 1) {
+            return getNode(((Element) parent).getElementsByTagName(nodePath[offset]).item(0), nodePath, offset + 1);
+        } else {
+            return ((Element) parent).getElementsByTagName(nodePath[offset]).item(0);
+        }
+    }
+
+    public static boolean childNodeExists(Node parent, String childNodeName) {
+        return ((Element) parent).getElementsByTagName(childNodeName).item(0) != null;
+    }
+
+    static Node getNode(Node root, String nodePath) {
+        String[] nodePathArr = nodePath.split("/");
+        return getNode(root, nodePathArr, 0);
+    }
+
+    /**
+     * Finds the node starting with the root and following the path. If the node is found it's inner text is returned,
+     * otherwise the default provided value.
+     * @param root
+     * @param nodePath
+     * @param defaultValue
+     * @return
+     */
+    public static String getNodeContentOrDefault(Node root, String nodePath, String defaultValue) {
```

Comment:
This method could potentially remove a lot of duplication, please use it on all applicable places.